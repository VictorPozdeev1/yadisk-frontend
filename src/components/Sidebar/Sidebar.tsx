import React, { FC } from "react";
import styles from "./Sidebar.module.css";

const Sidebar: FC = () => {
  return <div className={styles["Sidebar-wrapper"]}>Sidebar</div>;
};

export default Sidebar;
