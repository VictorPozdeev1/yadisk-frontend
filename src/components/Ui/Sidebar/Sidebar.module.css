.wrapper {
  /* border: 1px solid black; */
  width: 250px;
}
.active {
  color: #035fa2;
}
