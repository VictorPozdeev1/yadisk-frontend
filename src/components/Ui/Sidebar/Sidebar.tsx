import React, { FC } from "react";
import styles from "./Sidebar.module.css";

const Sidebar: FC<any> = ({ categories }) => {
  const content = categories
    ? categories.map((item: any) => (
        <div key={item.resource_id}>{item.name}</div>
      ))
    : null;

  return <div className={styles["wrapper"]}>{content}</div>;
};

export default Sidebar;
