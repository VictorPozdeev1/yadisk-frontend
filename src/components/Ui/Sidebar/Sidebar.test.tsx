import { render, screen } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import Sidebar from './Sidebar'

test('Sidebar component', () => {
    render(
        <BrowserRouter>
            <Sidebar
            />
        </BrowserRouter>
    )

})