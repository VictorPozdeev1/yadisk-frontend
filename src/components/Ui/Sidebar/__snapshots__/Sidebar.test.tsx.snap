// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component Sidebar snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="sidebar_test"
        style="display: flex; height: 100%; flex-direction: column; align-items: center; text-align: left; position: relative; gap: 27px; padding: 20px 12px 30px; min-width: 250px;"
      >
        <div
          data-testid="logo_test"
        >
          <img
            alt="Росатом"
            src="/images/logo.png"
            style="max-width: 181px;"
          />
        </div>
        <div
          style="display: flex; flex-direction: column; width: 100%;"
        >
          <div
            data-testid="toggle-el"
            style="width: 90%; background-color: rgb(255, 255, 255); border-radius: 7px;"
          >
            <a
              aria-current="page"
              class="active"
              data-testid="toggle-go-home-btn"
              href="/"
              style="padding: 12px 24px 12px 24px; width: 100%; display: block; text-decoration: none; color: rgb(3, 95, 162); cursor: pointer;"
            >
              Все документы
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="sidebar_test"
      style="display: flex; height: 100%; flex-direction: column; align-items: center; text-align: left; position: relative; gap: 27px; padding: 20px 12px 30px; min-width: 250px;"
    >
      <div
        data-testid="logo_test"
      >
        <img
          alt="Росатом"
          src="/images/logo.png"
          style="max-width: 181px;"
        />
      </div>
      <div
        style="display: flex; flex-direction: column; width: 100%;"
      >
        <div
          data-testid="toggle-el"
          style="width: 90%; background-color: rgb(255, 255, 255); border-radius: 7px;"
        >
          <a
            aria-current="page"
            class="active"
            data-testid="toggle-go-home-btn"
            href="/"
            style="padding: 12px 24px 12px 24px; width: 100%; display: block; text-decoration: none; color: rgb(3, 95, 162); cursor: pointer;"
          >
            Все документы
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
